frontend/src/plugins/errors.ts

```diff
@@ -0,0 +1,147 @@
+ */
+export function normalizeFetchingError(
+  error: unknown,
+  searchType: SupportedSearchType,
+  requestKind: RequestKind,
+  details?: Record<string, string>
+): FetchingError {
+  const fetchingError: FetchingError = {
+    requestKind,
+    details,
+    searchType,
+    code: ERR_UNKNOWN,
+  }
+
+  if (!axios.isAxiosError(error)) {
+    fetchingError.message = (error as Error).message
+    return fetchingError
+  }
+
+  // Otherwise, it's an AxiosError
+  if (isValidErrorCode(error.code)) {
+    fetchingError.code = error.code
+  }
+
+  if (error.response?.status) {
+    fetchingError.statusCode = error.response.status
+  }
+
+  const responseData = error?.response?.data
+
+  // Use the message returned by the API.
+  if (isDetailedResponseData(responseData)) {
+    fetchingError.message = responseData.detail as string
+  } else {
+    fetchingError.message = error.message
+  }
+
+  return fetchingError
+}
+
+/**
+ * Record network errors using the appropriate tool, as needed,
+ * based on response code, status, and request kind.
+ * @param error - The error to record
+ */
+function recordError(
+  context: Context,
+  originalError: unknown,
+  fetchingError: FetchingError
+) {
+  if (fetchingError.statusCode === 429) {
+    // These are more readily monitored via the Cloudflare dashboard.
+    return
+  }
+
+  if (
+    fetchingError.requestKind === "single-result" &&
+    fetchingError.statusCode === 404
+  ) {
+    /**
+     * Do not record 404s for single result requests because:
+     * 1. Plausible will already record them as resulting in a 404 page view
+     * 2. The Openverse API 404s on malformed identifiers, so there is no way
+     *    to distinguish between truly not found works and bad requests from
+     *    the client side.
+     * 3. There isn't much we can do other than monitor for an anomalously high
+     *    number of 404 responses from the frontend server that could indicate a frontend
+     *    implementation or configuration error suddenly causing malformed
+     *    identifiers to be used. Neither Sentry nor Plausible are the right tool
+     *    for that task. If the 404s are caused by an API issue, we'd see that in
+     *    API response code monitoring, where we can more easily trace the cause
+     */
+    return
+  }
+
+  if (process.client && fetchingError.code === "ERR_NETWORK") {
+    /**
+     * Record network errors in Plausible so that we can evaluate potential
+     * regional or device configuration issues, for which Sentry is not
+     * as good a tool. Additionally, the number of these events are trivial
+     * for Plausible, but do actually affect our Sentry quota enough that it
+     * is worth diverting them.
+     */
+    context.$sendCustomEvent("NETWORK_ERROR", {
+      requestKind: fetchingError.requestKind,
+      searchType: fetchingError.searchType,
+    })
+  } else {
+    context.$sentry.captureException(originalError, {
+      extra: { fetchingError },
+    })
+  }
+}
```

Comment:
``
```suggestion
    return
  } 
  
  // Send all other errors to Sentry
  context.$sentry.captureException(originalError, {
    extra: { fetchingError },
  })
}
```

Style nit: perhaps this would make adding future additional cases less error-prone, or avoid confusion if folks think there's some special relationship between the `process.client && fetchingError.code === "ERR_NETWORK"` errors and Sentry specifically.

Reply:
I went back and forth many times on this, and ultimately found both just as confusing, and went with a coin toss. I'll switch this, as any actual preference is better than my 50/50 non-choice!